include/pybind11/pybind11.h

```diff
@@ -466,7 +466,15 @@ class cpp_function : public function {
             }
             if (rec->def) {
                 std::free(const_cast<char *>(rec->def->ml_doc));
-                delete rec->def;
+                // Python 3.9.0 decref's these in the wrong order; rec->def
+                // If loaded on 3.9.0, let these leak (use Python 3.9.1 to fix)
+                // See https://github.com/python/cpython/pull/22670
+                #if !defined(PYPY_VERSION) && PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION == 9
+                    if (Py_GetVersion()[4] != '0')
```

Comment:
https://github.com/python/cpython/blob/04b8631d84a870dda456ef86039c1baf34d08500/Python/getversion.c is not extremely cheap, though, as it pastes together a bunch of strings, each time: https://github.com/python/cpython/blob/04b8631d84a870dda456ef86039c1baf34d08500/Python/getversion.c

Is that fine, or should we add this to `details()` or so?

It's probably fine, but just notifying, because this overhead will be there for all 3.9 versions, so also after 3.9.1 is out?

Reply:
I thought this was static, but I see that it isn't. We could avoid running it each time then.

Reply:
The question is if it's worth polluting the rest of the code with, though.

Oh, but it could be made static and shared with all other extension modules, ofc, since they would all be loaded into the same interpreter. We just need to make sure it's safe wrt ABI, I guess?

Reply:
Did you consider using `PY_MICRO_VERSION`? (I double-checked, it exists in Python 2.7 already, although it doesn't actually matter here, as long as you test `PY_MAJOR_VERSION` and `PY_MINOR_VERSION` first.)

Reply:
That's the compile time version, not the runtime version. Py_GetVersion grabs the version from when the running interpreter. So wheels work on any Python version.

Reply:
Oh, got it, thanks! Might be nice to add a comment after line 472, with that explanation.